asyncscope.md

```diff
@@ -2306,6 +2301,401 @@ template <async_scope_token Token, sender Sender>
+}
+```
+::::
+
+:::cmptable
+### Before
+```cpp
+namespace std::execution {
+
+template <class Token>
+concept async_scope_token =
+    copyable<Token> &&
+    requires(Token token) {
+        { token.try_associate() } -> same_as<bool>;
+        { token.disassociate() } noexcept -> same_as<void>;
+        { token.wrap(declval<@_test-sender_@>()) } -> sender_in<@_test-env_@>;
+    };
+
+struct simple_counting_scope::token {
+    template <sender Sender>
+    Sender&& wrap(Sender&& snd) const noexcept;
+
+    bool try_associate() const noexcept;
+
+    void disassociate() const noexcept;
+
+private:
+    simple_counting_scope* @_scope_@; // exposition only
+};
+
+struct counting_scope::token {
+    template <sender Sender>
+    sender auto wrap(Sender&& snd) const noexcept;
+    bool try_associate() const noexcept;
+    void disassociate() const noexcept;
+
+private:
+    counting_scope* @_scope_@; // @_exposition only_@
+};
+
+}
+```
+### After
+```cpp
+namespace std::execution {
+
+template <class Assoc>
+concept async_scope_assocation =
+    copyable<Assoc> &&
+    default_initializable<Assoc> &&
+    requires(Assoc assoc) {
+        { static_cast<bool>(assoc) } noexcept;
+    };
+
+template <class Token>
+concept async_scope_token =
+    copyable<Token> &&
+    requires(Token token) {
+        { token.try_associate() } -> async_scope_association;
+        { token.wrap(declval<@_test-sender_@>()) } -> sender_in<@_test-env_@>;
+    };
+
+struct simple_counting_scope::assoc {
+    explicit operator bool() const noexcept;
+
+private:
+    simple_counting_scope* @_scope_@; // exposition only
+};
+
+struct simple_counting_scope::token {
+    template <sender Sender>
+    Sender&& wrap(Sender&& snd) const noexcept;
+
+    assoc try_associate() const noexcept;
+
+private:
+    simple_counting_scope* @_scope_@; // exposition only
+};
+
+
+struct counting_scope::assoc {
+    explicit operator bool() const noexcept;
+
+private:
+    counting_scope* @_scope_@; // @_exposition only_@
+};
+
+struct counting_scope::token {
+    template <sender Sender>
+    sender auto wrap(Sender&& snd) const noexcept;
+
+    assoc try_associate() const noexcept;
+
+private:
+    counting_scope* @_scope_@; // @_exposition only_@
+};
+
+}
+```
+::::
+
+:::cmptable
+### Before
+```cpp
+namespace std::execution {
+
+template <class Alloc, async_scope_token Token, sender Sender>
+struct @_spawn-state_@ : @_spawn-state-base_@ {
+    using @_op-t_@ = connect_result_t<Sender, @_spawn-receiver_@>; // @_exposition only_@
+
+    @_spawn-state_@(Alloc a, Sender&& sndr, Token t)
+        : @_alloc_@(std::move(a)),
+          @_op_@(connect(std::move(sndr), @_spawn-receiver_@{this})),
+          @_token_@(std::move(t)) {}
+
+    void @_run_@() {
+        if (@_token_@.try_associate())
+            @_op_@.start();
+        else
+            @_destroy_@();
+    }
+    void @_complete_@() noexcept(is_nothrow_move_constructible_v<Token>) override {
```

Comment:
The token concept requires that tokens be no-throw move-constructible so we can leave this off.
```suggestion
    void @_complete_@() noexcept override {
```